pkg/provisioningv2/rke2/planner/planner.go

```diff
@@ -673,9 +689,125 @@ func addUserConfig(config map[string]interface{}, controlPlane *rkev1.RKEControl
 	return nil
 }
 
+// splitArgKeyVal takes a value and returns a pair (key, value) of the argument, or two empty strings if there was not
+// a parsed key/val.
+func splitArgKeyVal(val string, delim string) (string, string) {
+	if splitSubArg := strings.SplitN(val, delim, 2); len(splitSubArg) == 2 {
+		return splitSubArg[0], splitSubArg[1]
+	}
+	return "", ""
+}
+
+// getArgValue will search the passed in interface (arg) for a key that matches the searchArg. If a match is found, it
+// returns the value of the argument, otherwise it returns an empty string.
+func getArgValue(arg interface{}, searchArg string, delim string) string {
+	logrus.Tracef("getArgValue (searchArg: %s, delim: %s) type of %v is %T", searchArg, delim, arg, arg)
+	switch arg := arg.(type) {
+	case []interface{}:
+		logrus.Tracef("getArgValue (searchArg: %s, delim: %s) encountered interface slice %v", searchArg, delim, arg)
+		stringArr := convertInterfaceSliceToStringSlice(arg)
+		return getArgValue(stringArr, searchArg, delim)
+	case []string:
+		logrus.Tracef("getArgValue (searchArg: %s, delim: %s) found string array: %v", searchArg, delim, arg)
+		for _, v := range arg {
+			argKey, argVal := splitArgKeyVal(v, delim)
+			if argKey == searchArg {
+				return argVal
+			}
+		}
+	case string:
+		logrus.Tracef("getArgValue (searchArg: %s, delim: %s) found string: %v", searchArg, delim, arg)
+		argKey, argVal := splitArgKeyVal(arg, delim)
+		if argKey == searchArg {
+			return argVal
+		}
+	}
+	logrus.Tracef("getArgValue (searchArg: %s, delim: %s) did not find searchArg in: %v", searchArg, delim, arg)
+	return ""
+}
+
+// convertInterfaceSliceToStringSlice converts an input interface slice to a string slice by iterating through the
+// interface slice and converting each entry to a string using Sprintf.
+func convertInterfaceSliceToStringSlice(input []interface{}) []string {
+	var stringArr []string
+	for _, v := range input {
+		stringArr = append(stringArr, v.(string))
```

Comment:
If one of these values isn't a string for some reason, then this will panic. I liked your previous iteration that used `fmt.Sprintf("%v", v)`. That will always force `v` into a `string` regardless of its type.

Reply:
Converted to `fmt.Sprintf("%v", v)